### PR TITLE
sipcapture: add new capture_bad_msgs modparam

### PR DESCRIPTION
**Pull request by smititelu**
#### Pre-Submission Checklist

- [X] Commit message has the format required by CONTRIBUTING guide
- [X] Commits are split per component (core, individual modules, libs, utils, ...)
- [X] Each component has a single commit (if not, squash them into one commit)
- [X] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [X] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description

Enable the parsing of broken SIP messages.

@lbalaceanu is the original developer of this patch.

### CODE REVIEW

File: src/modules/sipcapture/sipcapture.c [hunk 1]
```diff
@@ -434,6 +436,64 @@ struct module_exports exports = {
 	destroy          /* module destroy function */
 };
 
+/*
+ * Find the first occurrence of find in s, where the search is limited to the
+ * first slen characters of s.
+ */
+char *strnstr(const char *s, const char *find, size_t slen)
```
Comment (henningw) [inline src/modules/sipcapture/sipcapture.c]: Please try to use the existing implementation:
src/core/str.h
130:char* _strnstr(const char *s, const char *find, size_t slen);

File: src/modules/sipcapture/sipcapture.c [hunk 2]
```diff
@@ -1946,30 +2033,35 @@ static int sip_capture(sip_msg_t *msg, str *_table,
...
-					contact.user.len = star_contact.len;
+				LM_DBG("error while parsing <Contact:> header\n");
+			}
+		} else {
+			cb = (contact_body_t*)msg->contact->parsed;
+			if (cb) {
+				if (cb->contacts) {
+					if (parse_uri(cb->contacts->uri.s, cb->contacts->uri.len, &contact) < 0) {
+						if (!parse_bad_msgs) {
+						   LOG(L_ERR, "ERROR: do_action: bad contact dropping"" packet\n");
```
Comment (henningw) [inline src/modules/sipcapture/sipcapture.c]: not needed "" probably from line wrapping

File: src/modules/sipcapture/sipcapture.c [hunk 3]
```diff
@@ -434,6 +436,64 @@ struct module_exports exports = {
...
+	end = msg->buf+msg->len;
+	tmp = strnstr(tmp, "Call-ID", (int)(end - tmp) );
+
+	if (tmp == NULL) {
+		LM_DBG("Bad msg callid not found\n");
+		EMPTY_STR(sco->callid);
+	} else {
+		hdr=pkg_malloc(sizeof(struct hdr_field));
+		if (unlikely(hdr==0)){
+			LOG(L_ERR, "ERROR:parse_headers: memory allocation error\n");
```
Comment (henningw) [inline src/modules/sipcapture/sipcapture.c]: consider removing this log statement and just use PKG_MEM_ERROR macro instead. The "parse_headers" prefix makes also not much sense in this context, I think.

**Comment by smititelu** [comment]
updated the commit accoding to the above comments.

**Comment by henningw** [comment]
Thanks you. Lets wait a bit if there are other comments from developers, otherwise it probably can be merged by the end of the week.

**Pull request merged**